### PR TITLE
CXP-14337: Global alert style update

### PR DESCRIPTION
**Pull request by larowlan**
- Updates global alert styling
- New icons
- New stories

### Desktop
Screenshot from 2022-05-18 14-45-57
Screenshot from 2022-05-18 14-45-49
### With close/title - mobile
Screenshot from 2022-05-18 14-45-29
### Mobile - no title/close
Screenshot from 2022-05-18 14-45-10
Screenshot from 2022-05-18 14-45-17

### CODE REVIEW

File: packages/alert/src/__tests__/alerts.test.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      value: {
+        href: 'http://127.0.0.1:8080/stuff/i-wish-i-had-known'
+      }
+    });
+    new Alerts(container.querySelector('.uq-alerts-global-container'));
+    await waitFor(() => {
+      expect(screen.getAllByRole('alert')).toHaveLength(2);
+    })
+    expect(container).toMatchSnapshot();
+  });
```
Comment (mstrelan) [inline packages/alert/src/__tests__/alerts.test.js:89]: I think we might need more cases here. Who says the pattern matching is working at all and not just returning the two alerts in the json file?

I also wondered if the pattern matching should ignore the scheme, e.g. someone puts http://www.uq.edu.au/ in Drupal but it's access on https, or vice versa (hopefully not).

File: packages/alert/src/js/alerts.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+      if (dismissedAtTimestamp > alert.dismissalIgnoreBefore) {
+        return false;
+      }
+      if (!alert.showOnPages.length) {
+        return true;
+      }
+      return alert.showOnPages.find(page => {
+        const quoted = `${page}`
+          .replace(new RegExp(`[.\\\\+?\\[^\\]$(){}=!<>|:\\-]`, 'g'), '\\$&');
```
Comment (mstrelan) [inline packages/alert/src/js/alerts.js:51]: :exploding_head:

File: packages/alert/src/__tests__/alerts.test.js (hunk 1 above)
Comment (mstrelan) [reply packages/alert/src/__tests__/alerts.test.js:89]: Probably also need to test the negate option.

**Review by mstrelan — COMMENTED** [review]
This looks awesome :star_struck:

File: packages/alert/src/__tests__/alerts.test.js (hunk 1 above)
Comment (larowlan) [reply packages/alert/src/__tests__/alerts.test.js:89]: :star_struck: yeah great points re the matching/tests
For the http vs https that gets a bit trickier, wouldn't mind punting that to the follow up that adds validation to the field in Drupal

File: packages/alert/src/js/alerts.js (hunk 2 above)
Comment (larowlan) [reply packages/alert/src/js/alerts.js:51]: this is basically `preq_quote` from PHP land, implemented in JavaScript, minus the quoting on `*` which we want to retain and switch to `.*`

File: packages/alert/src/__tests__/alerts.test.js (hunk 1 above)
Comment (mstrelan) [reply packages/alert/src/__tests__/alerts.test.js:89]: Fine by me. Ideally no one is using http.

Comment (larowlan) [reply packages/alert/src/__tests__/alerts.test.js]: pushed some more cases, and yep, didn't support negate at all, fixed

**Review by mstrelan — APPROVED** [review]

File: packages/alert/src/__tests__/alerts.test.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      expect(alerts[0]).not.toBeInTheDocument();
+    })
+    expect(container.body.innerHTML).toMatchSnapshot();
+    expect(parseInt(window.localStorage.getItem(`alert-dismissed-8da600f4-0a64-4df5-8bc5-a410d7496a26`))).toBeGreaterThanOrEqual(now);
+  });
+  it('Should match on partial URIs', async () => {
+    const container = renderFromString(headerMarkup);
+    Object.defineProperty(window, 'location', {
+      value: {
+        href: 'http://127.0.0.1:8080/stuff/i-wish-i-had-known'
```
Comment (hoffismo) [inline packages/alert/src/__tests__/alerts.test.js:82]: ❤️

**Review by hoffismo — APPROVED** [review]
Well done on the setup and Jest test @larowlan. Looks great.

**Review by Bcrunk — APPROVED** [review]

**Pull request merged**